oximeter/oximeter/src/schema.rs

```diff
@@ -0,0 +1,640 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2023 Oxide Computer Company
+
+//! Tools for working with schema for fields and timeseries.
+
+use crate::types::DatumType;
+use crate::types::FieldType;
+use crate::types::MetricsError;
+use crate::types::Sample;
+use crate::Metric;
+use crate::Target;
+use chrono::DateTime;
+use chrono::Utc;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::btree_map::Entry;
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use std::fmt::Write;
+use std::path::Path;
+
+/// The name and type information for a field of a timeseries schema.
+#[derive(
```

Comment:
Almost all of this positive diff was simply moved from `oximeter/db/src/lib.rs`.